cmd/query/app/query_parser.go

```diff
@@ -143,31 +160,125 @@ func (p *queryParser) parse(r *http.Request) (*traceQueryParameters, error) {
 	return traceQuery, nil
 }
 
-func (p *queryParser) parseTime(param string, r *http.Request) (time.Time, error) {
-	value := r.FormValue(param)
-	if value == "" {
-		if param == startTimeParam {
+func (p *queryParser) parseMetricsQueryParams(r *http.Request) (bqp metricsstore.BaseQueryParameters, err error) {
+	serviceNames := r.FormValue(servicesParam)
+	if serviceNames == "" {
+		return bqp, newParseError(errors.New("please provide at least one service name"), servicesParam)
+	}
+	bqp.ServiceNames = strings.Split(serviceNames, ",")
+
+	bqp.GroupByOperation, err = p.parseBool(r, groupByOperationParam)
+	if err != nil {
+		return bqp, err
+	}
+	bqp.SpanKinds, err = p.parseSpanKinds(r, spanKindsParam, defaultMetricsSpanKinds)
+	if err != nil {
+		return bqp, err
+	}
+	endTs, err := p.parseTime(r, endTsParam, msDuration)
+	if err != nil {
+		return bqp, err
+	}
+	lookback, err := p.parseDuration(r, lookbackParam, &msDuration, defaultMetricsQueryLookbackDuration)
+	if err != nil {
+		return bqp, err
+	}
+	step, err := p.parseDuration(r, stepParam, &msDuration, defaultMetricsQueryStepDuration)
+	if err != nil {
+		return bqp, err
+	}
+	ratePer, err := p.parseDuration(r, rateParam, &msDuration, defaultMetricsQueryRateDuration)
+	if err != nil {
+		return bqp, err
+	}
+	bqp.EndTime = &endTs
+	bqp.Lookback = &lookback
+	bqp.Step = &step
+	bqp.RatePer = &ratePer
+	return bqp, err
+}
+
+// parseTime parses the time parameter of an HTTP request that is represented the number of "units" since epoch.
+// If the time parameter is empty, the current time will be returned.
+func (p *queryParser) parseTime(r *http.Request, paramName string, units time.Duration) (time.Time, error) {
```

Comment:
as above, `units` are required here to support both micros and millis since epoch.